crates/net/network/src/manager.rs

```diff
@@ -926,6 +931,7 @@ where
             // ensure we still have enough budget for another iteration
             budget -= 1;
             if budget == 0 {
+                trace!(target: "net", budget=10, "exhausted network manager budget");
```

Comment:
```suggestion
                trace!(target: "net",
                    budget=10,
                    poll_duration_us=poll_duration_us,
                    max_healthy_poll_duration_us=max_healthy_poll_duration_us,
                    "exhausted network manager budget"
               );
```

as mentioned above, swarm can still block and starve other threads. so one iteration here might take a second, or it might take some microseconds.

even if we made the duration the budget, we would still not return from swarm if it blocks so let's aim at unblocking swarm (most pressingly the nested stream `NetworkState` https://github.com/paradigmxyz/reth/blob/29701f6486717514673e61b2b9b49418de7c4187/crates/net/network/src/state.rs#L398-L471).

I'd still like us to track the actual microseconds in this loop nonetheless.

at best, budget and max poll duration max are constants, but we can fix that nit at some later time point.